lib/ae_mdw_web/websocket/subscriptions.ex

```diff
@@ -48,41 +50,43 @@ defmodule AeMdwWeb.Websocket.Subscriptions do
     {:ok, :no_state}
   end
 
-  @spec subscribe(pid(), channel()) ::
+  @spec subscribe(pid(), version(), channel()) ::
           {:ok, [channel()]} | {:error, :invalid_channel | :already_subscribed}
-  def subscribe(pid, channel) do
-    with {:ok, channel_key} <- validate_subscribe(pid, channel) do
+  def subscribe(pid, version, channel) do
+    with {:ok, channel_key} <- validate_subscribe(pid, version, channel) do
       maybe_monitor(pid)
-      :ets.insert(@subs_pid_channel, {pid, channel})
+      :ets.insert(@subs_pid_channel, {pid, {channel_key, channel}})
       :ets.insert(@subs_channel_pid, {channel_key, pid})
 
       {:ok, subscribed_channels(pid)}
     end
   end
 
-  @spec unsubscribe(pid(), channel()) ::
+  @spec unsubscribe(pid(), version(), channel()) ::
           {:ok, [channel()]} | {:error, :invalid_channel | :not_subscribed}
-  def unsubscribe(pid, channel) do
-    with {:ok, channel_key} <- validate_unsubscribe(pid, channel) do
+  def unsubscribe(pid, version, channel) do
+    with {:ok, channel_key} <- validate_unsubscribe(pid, version, channel) do
       :ets.delete_object(@subs_channel_pid, {channel_key, pid})
-      :ets.delete_object(@subs_pid_channel, {pid, channel})
+      :ets.match_delete(@subs_pid_channel, {pid, {channel_key, :_}})
 
       {:ok, subscribed_channels(pid)}
     end
   end
 
-  @spec subscribers(channel()) :: [pid()]
-  def subscribers(channel) do
+  @spec subscribers(version(), channel()) :: [pid()]
+  def subscribers(version, channel) do
+    {:ok, channel_key} = channel_key(version, channel)
+
     @subs_channel_pid
-    |> :ets.match({channel, :"$1"})
+    |> :ets.match({channel_key, :"$1"})
     |> List.flatten()
   end
 
   @spec subscribed_channels(pid()) :: [channel()]
   def subscribed_channels(pid) do
     @subs_pid_channel
     |> :ets.match({pid, :"$1"})
-    |> List.flatten()
+    |> Enum.map(fn [{{_version, _channel_key}, channel}] -> channel end)
```

Comment:
This line works but it is confusing because the version is part of the channel_key.